algorithms/generated/scc.gsql

```diff
@@ -0,0 +1,174 @@
+CREATE DISTRIBUTED QUERY scc (INT iter = 500, INT iter_wcc = 5, INT top_k_dist) FOR GRAPH baidu{ //INT iter_end_trim = 3
+/* This query detects strongly connected components based on the following papers:
+   https://www.sandia.gov/~apinar/papers/irreg00.pdf
+   https://www.sciencedirect.com/science/article/pii/S0743731505000535
+   https://stanford-ppl.github.io/website/papers/sc13-hong.pdf
+
+   iter: number of iteration of the algorithm
+   iter_end_trim: stop trimming from this iteration, usually a small number(3 to 10)
+   iter_wcc: find wcc in this iteration; usually a small number(3 to 10)
+   top_k_dist: top k result in scc distribution
+
+   DISTRIBUTED QUERY mode is not supported.
+*/
+        TYPEDEF TUPLE <INT csize, INT num> cluster_num;
+        MapAccum<INT, INT> @@cluster_size_map, @@cluster_dist_map;
+        HeapAccum<cluster_num>(top_k_dist, csize DESC) @@cluster_dist_heap;
+        OrAccum @is_forward, @is_backward, @detached, @has_pos_indegree, @has_pos_outdegree, @wcc_active;
+        SumAccum<INT> @cid, @vid;
+        MinAccum<INT> @@min_vid, @wcc_id_curr, @wcc_id_prev;
+        SumAccum<STRING> @flag;
+        MapAccum<INT, MinAccum<INT>> @@f_cid_map, @@b_cid_map, @@n_cid_map, @@s_cid_map;
+        INT i = 1;
+        v_all = {User.*, Person.*};
+  
+        active = SELECT s
+                FROM v_all:s
+                ACCUM s.@vid = getvid(s),
+                      @@min_vid += getvid(s)
+                POST-ACCUM s.@cid = @@min_vid;
+  
+        WHILE active.size()>0 LIMIT iter DO
+                PRINT i;
+                // trim size 1 SCC
+                //IF i < iter_end_trim THEN
+                WHILE TRUE DO   
+                        tmp =  SELECT s
+                               FROM active:s -((link_to|link_to2|link_to3):e) -> :t
+                               WHERE t.@detached == FALSE AND s.@cid == t.@cid
+                               ACCUM s.@has_pos_outdegree = TRUE;
+  
+                        tmp =  SELECT s
+                               FROM active:s -((reverse_link_to|reverse_link_to2|reverse_link_to3):e) -> :t
+                               WHERE t.@detached == FALSE AND s.@cid == t.@cid
+                               ACCUM s.@has_pos_indegree = TRUE;
+                        trim_set = SELECT s
+                               FROM active:s
+                               WHERE s.@has_pos_indegree == FALSE OR s.@has_pos_outdegree == FALSE
+                               ACCUM s.@detached = TRUE,
+                                     s.@cid = s.@vid;
+                
+                        print trim_set.size();
```

Comment:
is this for debug?

Reply:
I want to show the user this information, then the user knows how the algorithm goes. Just like the modularity value for each iteration in Louvain